iroh-rpc-types/src/p2p.rs

```diff
@@ -144,6 +144,69 @@ pub struct LookupResponse {
     pub observed_addrs: Vec<Multiaddr>,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NetworkEventsRequest;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NetworkEventsResponse {
+    pub event: NetworkEvent,
+}
+
+use libp2p::gossipsub::{GossipsubMessage, MessageId, TopicHash};
+// TODO(ramfox): figure out better place for this or way to handle
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum NetworkEvent {
+    PeerConnected(PeerId),
+    PeerDisconnected(PeerId),
+    Gossipsub(GossipsubEvent),
+    CancelLookupQuery(PeerId),
+}
+
+// TODO(ramfox): figure out better place for this or way to handle
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum GossipsubEvent {
+    Subscribed {
+        peer_id: PeerId,
+        #[serde(with = "TopicHashDef")]
+        topic: TopicHash,
+    },
+    Unsubscribed {
+        peer_id: PeerId,
+        #[serde(with = "TopicHashDef")]
+        topic: TopicHash,
+    },
+    Message {
+        from: PeerId,
+        id: MessageId,
+        #[serde(with = "GossipsubMessageDef")]
+        message: GossipsubMessage,
+    },
+}
```

Comment:
To keep moving quickly, I just moved the definition of `NetworkEvent` & `GossipsubMessage` into `iroh-p2p-types`, since these are now being used by the rpc client when sending over a `NetworkEvent` stream & would cause a circular dependency if these definitions were not moved.

We can keep this as is and just re-export those structs in `iroh-p2p` (what this PR does currently).

Or, the other way I was considering, was to write mirrored structs here, `NetworkEventRPC` `GossipsubMessageRPC`, and then import them into `iroh-p2p` & have the `iroh-p2p` rpc methods be responsible for translating `NetworkEvent` and `GossipsubMessage` structs into their `iroh-rpc-types` counterparts before sending the actual rpc responses.

But I'm not sure which is more "right" from a rust-style perspective.

Reply:
I think the current approach of moving them here is very reasonable.  The alternative you describe seems to only involve more conversions without much obvious benefit.